Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs

```diff
@@ -32,17 +35,31 @@ public static class FileStreamResponseHelpers
             HttpContext httpContext,
             CancellationToken cancellationToken = default)
         {
-            if (state.RemoteHttpHeaders.TryGetValue(HeaderNames.UserAgent, out var useragent))
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, new Uri(state.MediaPath));
+            foreach (KeyValuePair<string, string> entry in state.RemoteHttpHeaders)
             {
-                httpClient.DefaultRequestHeaders.Add(HeaderNames.UserAgent, useragent);
+                request.Headers.Add(entry.Key, entry.Value);
+            }
+
+            if (httpContext.Request.Headers.TryGetValue(HeaderNames.Range, out StringValues values))
+            {
+                request.Headers.Add(HeaderNames.Range, values.ToArray());
             }
 
             // Can't dispose the response as it's required up the call chain.
-            var response = await httpClient.GetAsync(new Uri(state.MediaPath), cancellationToken).ConfigureAwait(false);
+            var response = await httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
             var contentType = response.Content.Headers.ContentType?.ToString() ?? MediaTypeNames.Text.Plain;
 
-            httpContext.Response.Headers[HeaderNames.AcceptRanges] = "none";
+            var headersToCopy = new List<string>() { HeaderNames.ContentLength, HeaderNames.ContentRange, HeaderNames.ContentDisposition, HeaderNames.ContentEncoding };
```

Comment:
This should either be a private static field or written out explicitly like below. You're unnecessarily creating a new list every time this method is called.

```
            var responseHeaders = response.Content.Headers;
            var headers = httpContext.Response.Headers;
            if (responseHeaders.ContentLength != null)
            {
                headers.ContentLength = responseHeaders.ContentLength;
            }

            if (responseHeaders.ContentEncoding.Count > 0)
            {
                headers.ContentEncoding = responseHeaders.ContentEncoding.ToArray();
            }

            if (responseHeaders.TryGetValues(HeaderNames.ContentRange, out var contentRange))
            {
                headers.ContentRange = contentRange.ToArray();
            }

            if (responseHeaders.TryGetValues(HeaderNames.ContentDisposition, out var contentDisposition))
            {
                headers.ContentDisposition = contentDisposition.ToArray();
            }

Reply:
Thanks for taking a look at the PR -- I've addressed your feedback.

I've also since made further progress and added support to re-write segment URIs within M3U playlist files so that clients can stream remote content directly (without needing to transcode). I think this should be sufficient to re-enable remote direct streams, and with local testing, it has shown to have made a day-and-night difference w.r.t. its performance & responsiveness.